### PR TITLE
🔥 feat: Add support for configuring TLS Min Version

### LINKED ISSUE #3239: 🔥 feat: Make tls.Config MinVersion configurable
As discussed in [PR #3201](https://github.com/gofiber/fiber/pull/3201) in [this comment](https://github.com/gofiber/fiber/pull/3201#discussion_r1863989536), it would be beneficial to make the `tls.Config` `MinVersion` configurable for the server listener and ACME. This enhancement will allow users to specify their desired minimum TLS version, such as TLS1.3.

Requester: @gaby

### PR DESCRIPTION
**Pull request by dozheiny**
This pull request introduces a new configuration option to set the minimum TLS version in the `listen.go` file. The most important changes include adding the `TLSMinVersion` field to the `ListenConfig` struct and updating the default configuration and relevant methods to use this new field.

### New TLS Configuration Option:

* [`listen.go`](diffhunk://#diff-8a6813f17ba53a53b4ca15776268eb3d32519867058e03351b87b7aea23b4978R53-R57): Added `TLSMinVersion` field to the `ListenConfig` struct to allow setting the minimum TLS version.
* [`listen.go`](diffhunk://#diff-8a6813f17ba53a53b4ca15776268eb3d32519867058e03351b87b7aea23b4978R136): Set the default value of `TLSMinVersion` to `tls.VersionTLS12` in the `listenConfigDefault` function.
* [`listen.go`](diffhunk://#diff-8a6813f17ba53a53b4ca15776268eb3d32519867058e03351b87b7aea23b4978R156-R159): Updated the `listenConfigDefault` function to ensure `TLSMinVersion` is set to `tls.VersionTLS12` if not provided.

### Code Updates to Use New Configuration:

* [`listen.go`](diffhunk://#diff-8a6813f17ba53a53b4ca15776268eb3d32519867058e03351b87b7aea23b4978L172-R182): Modified the `Listen` method to use the `TLSMinVersion` from the `ListenConfig` struct for the `tls.Config` object.

This commit will resolve #3239

### CODE REVIEW

**Comment by gaby** [comment]
@dozheiny Can you also update the documentation under the `docs`  directory.

To fix the struct issue, the `Makefile` in this repo has a `make betteralign` command that will fix it. https://github.com/gofiber/fiber/blob/main/Makefile#L58

File: listen.go [hunk 1]
```diff
@@ -50,6 +50,11 @@ type ListenConfig struct {
 	// Default: nil
 	TLSConfigFunc func(tlsConfig *tls.Config) `json:"tls_config_func"`
 
+	// TLSMinVersion allows to set tls minimum version.
+	//
+	// Default: VersionTLS12
+	TLSMinVersion uint16 `json:"tls_min_version"`
```
Comment (gaby) [inline listen.go:56]: Is `tls.Version12` not a type?

Comment (dozheiny) [reply listen.go:56]: No, it's not. crypto/tls use uint16 for configuring tls.
https://pkg.go.dev/crypto/tls#Config.MinVersion

We can create a type for TLS versions in Fiber, but this is not the best practice.

Comment (gaby) [reply listen.go:56]: Ok, cool

**Comment by dozheiny** [comment]
Should I change the default value of the TLS version to 1.3?

**Comment by gaby** [comment]
> Should I change the default value of the TLS version to 1.3?

V1.2 is fine as default. Not everything supports 1.3, and 1.2 is still active

**Comment by gaby** [comment]
@dozheiny Although we shouldn't allow the usage of TLS1.0 and TLS1.1

**Comment by dozheiny** [comment]
@gaby So I think creating a type for TLS versions is the best way.

**Comment by gaby** [comment]
@dozheiny You can compare using:

```go
if cfg.TLSMinVersion == 0 {
    cfg.TLSMinVersion = tls.VersionTLS12
}

if cfg.TLSMinVersion =! tls.VersionTLS12 || cfg.TLSMinVersion =! tls.VersionTLS13 {
    panic("Supported TLS versions: 1.2, 1.3")
}
```

Right? That way we dont have a custom type

**Comment by dozheiny** [comment]
@gaby But we break the application with panic.
Also if we raise the error, We change the output of functions.

**Comment by gaby** [comment]
> @gaby But we break the application with panic. Also if we raise the error, We change the output of functions.

That's fine cause its an invalid configuration. If you search for "panic" in the code base you will see we use it often for bad configs

**Comment by dozheiny** [comment]
@gaby Thanks for fixing lint issues and adding tests

**Review by gaby — APPROVED** [review]
LGTM

**Review by ReneWerner87 — APPROVED** [review]
LGTM

**Pull request merged**